README.md

```diff
@@ -1 +1,185 @@
-# algebra
\ No newline at end of file
+# algebra
+
+The Algebra Learning Project is an educational initiative aimed at providing foundational implementations of finite fields and elliptic curves. These mathematical structures are essential in various areas of modern cryptography and computational algebra.
+
+## Objectives
+
+- **Finite Fields**: Implement basic arithmetic operations and modular arithmetic for finite fields.
+- **Elliptic Curves**: Develop point addition, point doubling, and scalar multiplication for elliptic curves over finite fields.
+- **Learning Focus**: Emphasize understanding and practical application of algebraic concepts in cryptography.
+
+## Finite Field
+
+In this project, I focus on implementing a prime field and the arithmetic operations within it. This includes defining a prime field structure and supporting various arithmetic operations such as addition, subtraction, multiplication, and division, all within the context of a prime modulus.
+
+### Usage
+
+Clone `finite_field` and config `Cargo.toml`
+
+```jsx
+[dependencies]
+finite_field = { path = "../finite_field" }
+```
+
```

Comment:
Cloning the `finite_field` directory alone probably won't work. Also I don't think this section is necessary, you can remove this.
```suggestion
```